services/webnn/ort/graph_builder_ort.cc

```diff
@@ -2032,6 +2010,213 @@ GraphBuilderOrt::AddLinearOperation(const mojom::Linear& linear) {
+    ASSIGN_OR_RETURN(hidden_state,
+                     PrependReshape(hidden_state, {1, hidden_state_shape[0],
+                                                   hidden_state_shape[1]}));
+
+    cell_state = GetOperandNameById(lstm.cell_state_operand_id);
+    const std::vector<uint32_t>& cell_state_shape =
+        GetOperand(lstm.cell_state_operand_id).descriptor.shape();
+    CHECK_EQ(cell_state_shape.size(), 2u);
+    // Reshape the cellState into a 3-D tensor, since the LSTM of ONNX
+    // requires the initial_c shape to be [num_directions, batch_size,
+    // hidden_size]. For lstmCell, `num_directions` is equal to 1.
+    ASSIGN_OR_RETURN(cell_state,
+                     PrependReshape(cell_state, {1, cell_state_shape[0],
+                                                 cell_state_shape[1]}));
+  }
+  inputs.push_back(hidden_state.c_str());
+  inputs.push_back(cell_state.c_str());
+
+  std::string peephole_weight;
+  if (lstm.peephole_weight_operand_id.has_value()) {
+    peephole_weight =
+        GetOperandNameById(lstm.peephole_weight_operand_id.value());
+    if constexpr (std::is_same_v<LstmType, mojom::LstmCell>) {
+      // Reshape the peepholeWeight into a 2-D tensor, since the LSTM of ONNX
+      // requires the peephole shape to be [num_directions, 3*hidden_size]. For
+      // lstmCell, `num_directions` is equal to 1.
+      const std::vector<uint32_t>& peephole_weight_shape =
+          GetOperand(lstm.peephole_weight_operand_id.value())
+              .descriptor.shape();
+      CHECK_EQ(peephole_weight_shape.size(), 1u);
+      ASSIGN_OR_RETURN(
+          peephole_weight,
+          PrependReshape(peephole_weight, {1, peephole_weight_shape[0]}));
+    }
+  }
+  inputs.push_back(peephole_weight.c_str());
+
+  std::vector<ScopedOrtOpAttr> attributes;
+  attributes.reserve(3);
+  std::string direction = "forward";
+  if constexpr (std::is_same_v<LstmType, mojom::Lstm>) {
+    direction = GetRecurrentNetworkDirection(lstm.direction);
+  }
+  attributes.push_back(
+      model_editor_.CreateAttribute(/*name=*/"direction", direction.c_str()));
+
+  const std::vector<std::string> activations = GetRecurrentNetworkActivations(
+      lstm.activations, direction == "bidirectional");
+  std::vector<const char*> activations_c_str;
+  for (const auto& activation : activations) {
+    activations_c_str.push_back(activation.c_str());
+  }
+  attributes.push_back(model_editor_.CreateAttribute(
+      /*name=*/"activations", activations_c_str));
+
+  attributes.push_back(model_editor_.CreateAttribute(
+      /*name=*/"hidden_size", base::checked_cast<int64_t>(hidden_size)));
+
+  // TODO(https://github.com/shiyi9801/chromium/issues/195): Support ifgo
+  // layout.
+  if (lstm.layout != mojom::LstmWeightLayout::kIofg) {
+    return NewNotSupportedError(
+        "[WebNN] The lstm weight layout (ifgo) is not supported.");
+  }
+
+  std::string output_Y, output_Y_h, output_Y_c;
```

Comment:
variable use lower-case name. How about 
```suggestion
  std::string output, output_hidden, output_cell;
```